$OpenBSD: patch-src_dh-assistant-view_c,v 1.2 2009/10/06 15:24:47 ajacoutot Exp $

From upstream SVN rev 1265:
Use file: URIs instead of local paths, fixes use with more recent webkit
versions.

From GNOME bugzilla #588655
g_mapped_file_free() has been deprecated in favor of g_mapped_file_unref().

--- src/dh-assistant-view.c.orig	Fri Nov 28 16:56:04 2008
+++ src/dh-assistant-view.c	Tue Oct  6 17:16:55 2009
@@ -169,6 +169,7 @@ assistant_view_set_link (DhAssistantView *view,
         gsize                length;
         gchar               *key;
         gsize                key_length;
+        gsize                offset = 0;
         const gchar         *start;
         const gchar         *end;
 
@@ -203,7 +204,10 @@ assistant_view_set_link (DhAssistantView *view,
                 return;
         }
 
-        file = g_mapped_file_new (filename, FALSE, NULL);
+        if (g_str_has_prefix (filename, "file://"))
+            offset = 7;
+        
+        file = g_mapped_file_new (filename + offset, FALSE, NULL);
         if (!file) {
                 g_free (filename);
                 return;
@@ -326,7 +330,7 @@ assistant_view_set_link (DhAssistantView *view,
                  * anchor links are handled internally in webkit.
                  */
                 tmp = g_path_get_dirname (filename);
-                base = g_strconcat ("file://", tmp, "/fake", NULL);
+                base = g_strconcat (tmp, "/fake", NULL);
                 g_free (tmp);
 
                 webkit_web_view_load_html_string (
@@ -340,7 +344,11 @@ assistant_view_set_link (DhAssistantView *view,
                 webkit_web_view_open (WEBKIT_WEB_VIEW (view), "about:blank");
         }
 
+#if GLIB_CHECK_VERSION(2,21,3)
+        g_mapped_file_unref (file);
+#else
         g_mapped_file_free (file);
+#endif
         g_free (filename);
 }
 
